codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpBoundProtocolGenerator.kt

```diff
@@ -696,6 +723,19 @@ private class ServerHttpBoundProtocolTraitImplGenerator(
         serverRenderUriPathParser(this, operationShape)
         serverRenderQueryStringParser(this, operationShape)
 
+        if (noInputs && protocol.serverContentTypeCheckNoModeledInput()) {
+            val emptyBody = if (parser != null) { "body.is_empty()" } else { "true" }
+            rustTemplate(
+                """
+                if $emptyBody {
```

Comment:
Clippy does not flag this `if true`?

I'd rewrite this to avoid it. You can use `conditionalBlock("if body.is_empty() {", "}", conditional = parser != null) { ... }` for this purpose.

Reply:
Thanks!